cmd/uhc/cluster/list/list.go

```diff
@@ -0,0 +1,192 @@
+	managed   bool
+}
+
+var Cmd = &cobra.Command{
+	Use:   "list [MANAGED | UNMANAGED] ",
+	Short: "List clusters",
+	Long:  "List clusters. Defaults to managed.",
+	Run:   run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.BoolVar(
+		&args.debug,
+		"debug",
+		false,
+		"Enable debug mode.",
+	)
+	flags.StringArrayVar(
+		&args.parameter,
+		"parameter",
+		nil,
+		"Query parameters to add to the request. The value must be the name of the "+
+			"parameter, followed by an optional equals sign and then the value "+
+			"of the parameter. Can be used multiple times to specify multiple "+
+			"parameters or multiple values for the same parameter.",
+	)
+	flags.BoolVar(
+		&args.managed,
+		"managed",
+		false,
+		"Filter unmanaged clusters",
+	)
+}
+func run(cmd *cobra.Command, argv []string) {
+
+	pageSize := 100
+	pageIndex := 1
+
+	// Load the configuration file:
+	cfg, err := config.Load()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't load config file: %v\n", err)
+		os.Exit(1)
+	}
+	if cfg == nil {
+		fmt.Fprintf(os.Stderr, "Not logged in, run the 'login' command\n")
+		os.Exit(1)
+	}
+
+	// Check that the configuration has credentials or tokens that haven't have expired:
+	armed, err := config.Armed(cfg)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't check if tokens have expired: %v\n", err)
+		os.Exit(1)
+	}
+	if !armed {
+		fmt.Fprintf(os.Stderr, "Tokens have expired, run the 'login' command\n")
+		os.Exit(1)
+	}
+
+	// Create the connection:
+	logger, err := util.NewLogger(args.debug)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't create logger: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Create the connection, and remember to close it:
+	connection, err := client.NewConnectionBuilder().
+		Logger(logger).
+		TokenURL(cfg.TokenURL).
+		Client(cfg.ClientID, cfg.ClientSecret).
+		Scopes(cfg.Scopes...).
+		URL(cfg.URL).
+		User(cfg.User, cfg.Password).
+		Tokens(cfg.AccessToken, cfg.RefreshToken).
+		Insecure(cfg.Insecure).
+		Build()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't create connection: %v\n", err)
+		os.Exit(1)
+	}
+	defer connection.Close()
+
+	// Get the client for the resource that manages the collection of clusters:
+	collection := connection.ClustersMgmt().V1().Clusters()
+
+	for {
+		// Fetch the next page:
+		response := getResponse(collection, args.managed, args.parameter, pageSize, pageIndex)
+
+		// Display the fetched page:
+		if response != nil {
+			response.Items().Each(func(cluster *v1.Cluster) bool {
+				fmt.Printf("ID: %s - Name: %s.%s\n", cluster.ID(), cluster.Name(), cluster.DNS().BaseDomain())
+				return true
+			})
+			// If the number of fetched results is less than requested, then
+			// this was the last page, otherwise process the next one:
+			if response.Size() < pageSize {
+				break
+			}
+			pageIndex++
+		} else {
+			break
+		}
+	}
+}
+
+func getResponse(collection *v1.ClustersClient, managed bool, parameter []string, pageSize int, pageIndex int) *v1.ClustersListResponse {
+
```

Comment:
This can be simplified creating the request outside of the `if`, with the common parameters, and then you can conditionally add what you need:

```go
// First create the basic request:
listRequest := collection.List().
        Size(pageSize).
        Page(pageIndex)

// Then add, conditionally what you need:
if managed {
        listRequest.Search("managed = 'true'")
}

// Add more conditional things (parameters should be compatible with managed):
if len(parameter) > 0 {
        listRequest....
}

// Finally send the request:
listResponse, err := listRequest.Send()
if err != nil {
         ....
}
```

Note also that requests can be re-used, so you could create the request with the basic parameters outside of this `getResponse` function, and then modify and send it multiple times.